### DISCUSSION TITLE
Concourse task to run script can't find the linux command sudo

### DISCUSSION THREAD

**Discussion by martijnvandenboom** [concourse/concourse#8434] [Help & Support]
Dear reader,

Currently I am studying on concourse tasks running scripts.

This is my script:

```
#!/bin/sh
 
whoami
 
env
 
sudo kubectl --context kind-kind get pods --all-namespaces

```

This is my YAML file:

```
---
platform: linux

image_resource:
  type: docker-image
  source: {repository: busybox}

inputs:
  - name: task-scripts

run:
  path: /bin/sh
  args: ["./task-scripts/task_status-cf.sh"]

```

This is the command:

`$ fly -t tutorial e -c task_status-cf.yml`

This is the output:

```
uploading task-scripts done
executing build 144 at http://localhost:8080/builds/144
initializing
initializing check: image
selected worker: e9f4ac7368bb
INFO: found existing resource cache

selected worker: e9f4ac7368bb
running /bin/sh ./task-scripts/task_status-cf.sh
root
USER=root
SHLVL=1
HOME=/root
PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
PWD=/tmp/build/e55deab7
ATC_EXTERNAL_URL=http://localhost:8080
./task-scripts/task_status-cf.sh: line 7: sudo: not found
failed
```

I would very much like to improve my troubleshooting skills.
Can anybody help me?
Does anyone have a tip or an idea to resolve this?

**Comment by marco-m** [comment]
Hello,

inside the container you are already running as root, see the output of whoami. So, you do not need sudo. 
Or more exactly: sudo will not allow you to do anything more than what you are already able to do without.

To troubleshoot the script on your local machine, you need to run it from within a Docker container (using the same image as the one you set in the Concourse task), to reproduce as close as possible the Concourse environment.

Lastly, I have the feeling that using kubectl from the script is not the best way to understand how Concourse and containers work, because it will run in a container with a network configuration that is not what you expect, and you will be just confused.

I suggest to pick a simpler command to start with.

**Reply by martijnvandenboom** [reply]
@marco-m Hello Marco, 

First let me thank you for taking a look into my issue. So grateful.
The actual issue was that the container didn't have much of the standard Linux commands on board.
I didn't have much experience with creating custom images so I tried to find one on: hub.docker.com
Even there the standard containers couldn't do the trick I was going for.
I ended up creating a repository there and uploaded my own custom made container.

A good resource is: [create your own custom made container](https://dev.to/ianknighton/building-a-custom-docker-image-32bp)

So basically now I decide which commands are available and my solution works, pretty cool.
Thanks so much for your effort to reply to me.

Sincerely
